test/core/surface/concurrent_connectivity_test.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+
+#include <grpc/grpc.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/thd.h>
+
+#define NUM_THREADS 100
+static grpc_channel* channels[NUM_THREADS];
+static grpc_completion_queue* queues[NUM_THREADS];
+
+void create_loop_destroy(void* actually_an_int) {
+  int thread_index = (int)(actually_an_int);
+  for (int i = 0; i < 10; ++i) {
+    grpc_completion_queue* cq = grpc_completion_queue_create(NULL);
+    grpc_channel* chan = grpc_insecure_channel_create("localhost", NULL, NULL);
+
+    channels[thread_index] = chan;
```

Comment:
If I'd known this was going to go straight into becoming a test, I'd have cleaned up these lines. :expressionless: